server/gui/main.py

```diff
@@ -0,0 +1,212 @@
+# flake8: noqa F403, F405
+import sys
+from os.path import splitext, basename
+
+from cefpython3 import cefpython as cef
+from PyQt5.QtCore import *
+from PyQt5.QtWidgets import *
+
+from server.gui.browser import CefWidget, CefApplication
+from server.gui.cxg_server import cellxgeneServer, DataLoadWorker, ServerRunWorker
+from server.gui.utils import WINDOWS, LINUX, MAC
+
+# Configuration
+# TODO remember this or calculate it?
+WIDTH = 1024
+HEIGHT = 768
+
+
+# noinspection PyUnresolvedReferences
+class MainWindow(QMainWindow):
+    def __init__(self):
+        super(MainWindow, self).__init__(None)
+        self.thread_pool = QThreadPool()
+        self.cef_widget = None
+        self.data_widget = None
+        self.server = cellxgeneServer(self)
+        self.server.setup_app()
+        self.runServer()
+        self.setWindowTitle("cellxgene")
+
+        # Strong focus - accepts focus by tab & click
+        self.setFocusPolicy(Qt.StrongFocus)
+        self.setupLayout()
+        self.setupMenu()
+
+    def setupLayout(self):
```

Comment:
Python style question: I thought method names should follow function naming conventions, ie, snake case?   Reference [PEP 8 Method Names and Instance Variables](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables).   I'm fine if you have a reason for deviating - mostly trying to learn.

I see that PyQt uses this convention, so my guess is that the practice is to conform to that?

Reply:
PyQt is copying C++ style I believe, and there are a few methods in the classes I'm building that require that. In the interest of consistency if a class had at least one method in camel case, then I made all methods have camel case. Otherwise I used snake case. (see "A Foolish Consistency is the Hobgoblin of Little Minds" section of the PEP 8 link) I'm open to alternatives, I'm not sure I made the best choice for when to apply the rules and when not to.